packages/adapter-drizzle/src/lib/mysql.ts

```diff
@@ -252,21 +280,91 @@ export function MySqlDrizzleAdapter(
           )
         )
     },
+    async getAccount(providerAccountId, provider) {
+      return await client
+        .select()
+        .from(accountsTable)
+        .where(
+          and(
+            eq(accountsTable.providerAccountId, providerAccountId),
+            eq(accountsTable.provider, provider)
+          )
+        )
+        .then((res) => (res[0] as AdapterAccount) ?? null)
+    },
+    async createAuthenticator(data) {
+      await client
+        .insert(mysqlAuthenticatorsTable)
+        .values({ ...data, id: crypto.randomUUID() })
+
+      return await client
+        .select()
+        .from(authenticatorsTable)
+        .where(eq(authenticatorsTable.credentialID, data.credentialID))
+        .then((res) => fromDBAuthenticator(res[0]) ?? null)
+    },
+    async getAuthenticator(credentialID) {
+      const authenticator = await client
+        .select()
+        .from(authenticatorsTable)
+        .where(eq(authenticatorsTable.credentialID, credentialID))
+        .then((res) => (res.length ? fromDBAuthenticator(res[0]) : null))
+      return authenticator ? authenticator : null
+    },
+    async listAuthenticatorsByUserId(userId) {
+      return await client
+        .select()
+        .from(authenticatorsTable)
+        .where(eq(authenticatorsTable.userId, userId))
+        .then((res) => res.map(fromDBAuthenticator))
+    },
+    async updateAuthenticatorCounter(credentialID, newCounter) {
+      await client
+        .update(authenticatorsTable)
+        .set({ counter: newCounter })
+        .where(eq(authenticatorsTable.credentialID, credentialID))
+
+      return await client
+        .select()
+        .from(authenticatorsTable)
+        .where(eq(authenticatorsTable.credentialID, credentialID))
+        .then((res) => fromDBAuthenticator(res[0]) ?? null)
+    },
+  }
+}
+
+type DrizzleAuthenticator = Required<
+  DefaultMySqlSchema["authenticatorsTable"]["$inferInsert"]
```

Comment:
use official `InferInsertModel` helper instead of accessing internal types directly?